From 234914c6dd1059dbb1eec26f1953ad22259720c1 Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Tue, 8 Apr 2014 10:45:14 +0530
Subject: [PATCH 14/39] ASoC: soc-fw: Make widget load independent of widget
 controls

Issue: GMINL-3679

There can be widgets without any controls, But it should be still
be able to load the widget so move the code to make it
independent of widget controls code.

Change-Id: Idbfb29270e1242e5dfa987cb18c528c0c2063ef3
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/soc-fw.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index 2de4b1f..2f610d9 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -1147,9 +1147,6 @@ static int soc_fw_dapm_widget_create(struct soc_fw *sfw,
 			ret = -ENOMEM;
 			goto hdr_err;
 		}
-		ret = soc_fw_widget_load(sfw, &widget);
-		if (ret < 0)
-			goto hdr_err;
 		break;
 	case SOC_CONTROL_TYPE_ENUM:
 	case SOC_CONTROL_TYPE_ENUM_EXT:
@@ -1162,9 +1159,6 @@ static int soc_fw_dapm_widget_create(struct soc_fw *sfw,
 			ret = -ENOMEM;
 			goto hdr_err;
 		}
-		ret = soc_fw_widget_load(sfw, &widget);
-		if (ret < 0)
-			goto hdr_err;
 		break;
 	default:
 		dev_err(sfw->dev, "ASoC: invalid widget control type %d:%d:%d\n",
@@ -1176,6 +1170,10 @@ static int soc_fw_dapm_widget_create(struct soc_fw *sfw,
 	}
 
 widget:
+	ret = soc_fw_widget_load(sfw, &widget);
+		if (ret < 0)
+			goto hdr_err;
+
 	ret = snd_soc_dapm_new_controls(dapm, &widget, 1);
 	if (ret < 0) {
 		dev_err(sfw->dev, "ASoC: failed to create widget %s controls\n",
-- 
1.9.1

